rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/TestFileDownloader.java

```diff
@@ -38,52 +37,57 @@
 {
   private static final Log log = LogFactory.getLog(TestFileDownloader.class);
 
-  private static final String testDirectoryPrefix = System.getProperty("java.io.tmpdir") + "/TestFileDownloader/";
-  private static final String testDirectory = testDirectoryPrefix + "dir0";
-  private Configuration conf;
+  private static final String TEST_CACHE_DIR_PREFIX = BookKeeperTestUtils.getTestCacheDirPrefix("TestFileDownloader");
+  private static final int TEST_MAX_DISKS = 1;
+
+  private final Configuration conf = new Configuration();
 
   @BeforeMethod
   public void setUp() throws Exception
   {
-    conf = new Configuration();
-    CacheConfig.setCacheDataDirPrefix(conf, testDirectoryPrefix + "dir");
-    CacheConfig.setMaxDisks(conf, 1);
-    Files.createDirectories(Paths.get(testDirectory, CacheConfig.getCacheDataDirSuffix(conf)));
+    CacheConfig.setCacheDataDirPrefix(conf, TEST_CACHE_DIR_PREFIX);
+
+    BookKeeperTestUtils.createCacheParentDirectories(CacheConfig.getCacheDirPrefixList(conf), TEST_MAX_DISKS);
```

Comment:
Lets pass the conf object to removeCacheParentDirectories and createCacheParentDirectories and inside those tho methods get the cache directory prefix from the conf object rather than passing the prefix to the methods

Reply:
@abhishekdas99 Addressed.